src/renderers/dom/shared/DOMProperty.js

```diff
@@ -13,10 +13,26 @@
 
 var invariant = require('fbjs/lib/invariant');
 
+// These attributes should be all lowercase to allow for
+// case insensitive checks
+var RESERVED_PROPS = {
+  children: true,
+  dangerouslysetinnerhtml: true,
+  autofocus: true,
+  defaultvalue: true,
+  defaultchecked: true,
+  innerhtml: true,
```

Comment:
This is slightly confusing because if I use it, it says:

>Warning: Invalid DOM property `innerhtml`. Did you mean `innerHTML`?

But then if I try `innerHTML`, of course it isn’t supported:

>Warning: Directly setting property `innerHTML` is not permitted. For more information, lookup documentation on `dangerouslySetInnerHTML`.

It seems better if the same warning was displayed immediately (or if we just got a generic "unknown property" warning for wrong casing like `innerhtml`).